src/Orchard.Web/Modules/Orchard.ContentTypes/Controllers/AdminController.cs

```diff
@@ -142,17 +142,42 @@ public class AdminController : Controller, IUpdateModel {
 
             if (contentTypeDefinition == null) return HttpNotFound();
 
+            //Grouping Tabs > Cards > Shapes
             var grouped = _placementService.GetEditorPlacement(id)
                 .OrderBy(x => x.PlacementInfo.GetPosition(), new FlatPositionComparer())
                 .ThenBy(x => x.PlacementSettings.ShapeType)
                 .Where(e => e.PlacementSettings.Zone == "Content")
-                .GroupBy(x => x.PlacementInfo.GetTab())
-                .ToDictionary(x => x.Key, y => y.ToList());
-
-            var content = grouped.ContainsKey("") ? grouped[""] : new List<DriverResultPlacement>();
-            var listPlacements = grouped.Values.SelectMany(e => e).ToList();
-
-            grouped.Remove("");
+                .GroupBy(g => g.PlacementInfo.GetTab() + "%" + g.PlacementInfo.GetCard())
+                .Select(x =>
+                    new Card {
```

Comment:
This code should be next to `GetTab()`, not in the controller. And I have the feeling that this code is more complex than it should be.

Reply:
I my latest commit I added a bunch of comments to clarify what we are doing there. We The code can be simplified slightly (e.g. by removing the `Select` statements and have the `GroupBy` statements return the objects) but For now I kept it like this for clarity. Could you clarify the changes you would like here?